dist/templates/k8s.ovn.org_egressfirewalls.yaml.j2

```diff
@@ -0,0 +1,90 @@
+      properties:
+        apiVersion:
+          description: 'APIVersion defines the versioned schema of this representation of an object. Servers should convert recognized schemas to the latest internal value, and may reject unrecognized values. More info: https://git.k8s.io/community/contributors/devel/sig-architecture/api-conventions.md#resources'
+          type: string
+        kind:
+          description: 'Kind is a string value representing the REST resource this object represents. Servers may infer this from the endpoint the client submits requests to. Cannot be updated. In CamelCase. More info: https://git.k8s.io/community/contributors/devel/sig-architecture/api-conventions.md#types-kinds'
+          type: string
+        metadata: 
+          type: object
+          properties:
+            name:
+              type: string
+              pattern: ^default$
+        spec:
+          description: Specification of the desired behavior of EgressFirewall.
+          properties:
+            egress:
+              description: a collection of egress firewall rule objects
+              items:
+                description: EgressFirewallRule is a single egressfirewall rule object
+                properties:
+                  ports:
+                    description: ports mark what ports and protocols for the rule to affect
+                    items:
+                      description: EgressFirewallPort specifies the port to allow or deny traffic to
+                      properties:
+                        port:
+                          description: port that the traffic must match
+                          format: int32
+                          type: integer
+                        protocol:
+                          description: protocol (tcp, udp, sctp) that the traffic must match.
+                          pattern: ^tcp|udp|sctp$
+                          type: string
+                      type: object
+                    type: array
+                  to:
+                    description: to is one or more targes that traffic is allowed/denied to
+                    properties:
+                      cidrSelector:
+                        description: cidrSelector is the CIDR range to allow/deny traffic to. If this is set, dnsName must be unset.
+                        type: string
+                    type: object
+                  type:
+                    description: type marks this as an "Allow" or "Deny" rule
+                    pattern: ^Allow|Deny$
+                    type: string
+                required:
+                - to
+                - type
+                type: object
+              type: array
+          required:
+          - egress
+          type: object
+        status:
+          description: Observed status of EgressFirewall
+          type: object
+      required:
+      - spec
+      type: object
+  version: v1
```

Comment:
FYI Looks like per link  https://kubernetes.io/docs/reference/generated/kubernetes-api/v1.18/#customresourcedefinitionspec-v1beta1-apiextensions-k8s-io `version` is deprecated and that use of `versions` is encouraged. Find below text snippet from above link:
```
version is the API version of the defined custom resource. The custom resources are served under `/apis/<group>/<version>/...`. Must match the name of the first item in the `versions` list if `version` and `versions` are both specified. Optional if `versions` is specified. Deprecated: use `versions` instead.
```

Reply:
@vjayaramrh this is file generated from the controller-gen tools

Reply:
@JacobTanenbaum Thanks for the info